src/test/java/net/satago/tapestry5/jpa/test/CommitCounter.java

```diff
@@ -0,0 +1,54 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.satago.tapestry5.jpa.test;
+
+import java.util.Date;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.PostPersist;
+import javax.persistence.PostUpdate;
+
+import net.satago.tapestry5.jpa.EntityTransactionManager;
+import net.satago.tapestry5.jpa.test.entities.VersionedThing;
+
+public class CommitCounter {
+
+	@Inject
+	private EntityManager entityManager;
+
+    @Inject
+    private EntityTransactionManager transactionManager;
+
+	@PostPersist
+	@PostUpdate
+	private void updateVersion(Object entity) {
+        transactionManager.runInTransaction(null, new Runnable()
+        {
+            @Override
+            public void run()
+            {
+                VersionedThing versionedThing = entityManager.find(VersionedThing.class, 1);
+                if (versionedThing == null)
+                    versionedThing = new VersionedThing();
+                versionedThing.setId(1);
+                versionedThing.setLastTouched(new Date());
+
+                entityManager.merge(versionedThing);
```

Comment:
FYI

I'm trying to back port these tests to master and this bit doesn't work (at least with Hibernate) throwing a ConcurrentModificationException from Hibernate internals. Here's what I've found:
>To avoid conflicts with the original database operation that fires the entity lifecycle event (which is still in progress) callback methods should not call EntityMan­ager or Query methods and should not access any other entity objects.
>
>If a callback method throws an exception within an active transaction, the transaction is marked for rollback and no more callback methods are invoked for that operation.

https://www.objectdb.com/java/jpa/persistence/event#Implementation_Restrictions_

This is exactly what's happening here. I'm going to rewrite this without EntityManager.